packages/forwarder-helper/src/forwarder_helper_factory.ts

```diff
@@ -22,4 +31,220 @@ export const forwarderHelperFactory = {
         const helper = new ForwarderHelperImpl(config);
         return helper;
     },
+    /**
+     * Given a desired makerAsset and SRA url, get a ForwarderHelper
+     * @param   makerAssetData      An array of objects conforming to SignedOrder. Each order should specify the same makerAssetData and takerAssetData
+     * @param   sraUrl              A url pointing to an SRA v2 compliant endpoint.
+     * @param   rpcUrl              A url pointing to an ethereum node.
+     * @param   networkId           The ethereum networkId, defaults to 1 (mainnet).
+     * @return  A ForwarderHelper, see type for definition
+     */
+    async getForwarderHelperForMakerAssetDataAsync(
+        makerAssetData: string,
+        takerAddress: string,
+        sraUrl: string,
+        rpcUrl?: string,
+        networkId: number = 1,
+    ): Promise<ForwarderHelper> {
+        assert.isHexString('makerAssetData', makerAssetData);
+        assert.isETHAddressHex('takerAddress', takerAddress);
+        assert.isWebUri('sraUrl', sraUrl);
+        if (!_.isUndefined(rpcUrl)) {
+            assert.isWebUri('rpcUrl', rpcUrl);
+        }
+        assert.isNumber('networkId', networkId);
+        // create provider
+        const providerEngine = new Web3ProviderEngine();
+        if (!_.isUndefined(rpcUrl)) {
+            providerEngine.addProvider(new RPCSubprovider(rpcUrl));
+        }
+        providerEngine.start();
+        // create contract wrappers given provider and networkId
+        const contractWrappers = new ContractWrappers(providerEngine, { networkId });
+        // find ether token asset data
+        const etherTokenAddressIfExists = contractWrappers.etherToken.getContractAddressIfExists();
+        if (_.isUndefined(etherTokenAddressIfExists)) {
+            throw new Error(ForwarderHelperFactoryError.NoEtherTokenContractFound);
+        }
+        const etherTokenAssetData = assetDataUtils.encodeERC20AssetData(etherTokenAddressIfExists);
+        // find zrx token asset data
+        let zrxTokenAssetData: string;
+        try {
+            zrxTokenAssetData = contractWrappers.exchange.getZRXAssetData();
+        } catch (err) {
+            throw new Error(ForwarderHelperFactoryError.NoZrxTokenContractFound);
+        }
+        // get orderbooks for makerAsset/WETH and ZRX/WETH
+        const sraClient = new HttpClient(sraUrl);
+        const orderbookRequests = [
+            { baseAssetData: makerAssetData, quoteAssetData: etherTokenAssetData },
+            { baseAssetData: zrxTokenAssetData, quoteAssetData: etherTokenAssetData },
+        ];
+        const requestOpts = { networkId };
+        // TODO: try catch these requests and throw a more domain specific error
+        const [makerAssetOrderbook, zrxOrderbook] = await Promise.all(
+            _.map(orderbookRequests, request => sraClient.getOrderbookAsync(request, requestOpts)),
+        );
+        // validate orders and find remaining fillable from on chain state or sra api
```

Comment:
we potentially want to filter out orders where the taker is not null address (i.e. not an open order)